core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -472,59 +485,66 @@ public void addSplits(String tableName, SortedSet<Text> splits)
     ClientTabletCache tabLocator = ClientTabletCache.getInstance(context, tableId);
 
     SortedSet<Text> splitsTodo = new TreeSet<>(splits);
-    ExecutorService executor =
-        context.threadPools().getPoolBuilder("addSplits").numCoreThreads(16).build();
 
-    try {
-      while (!splitsTodo.isEmpty()) {
+    while (!splitsTodo.isEmpty()) {
 
-        tabLocator.invalidateCache();
+      tabLocator.invalidateCache();
 
-        Map<KeyExtent,List<Text>> tabletSplits =
-            mapSplitsToTablets(tableName, tableId, tabLocator, splitsTodo);
+      Map<KeyExtent,List<Text>> tabletSplits =
+          mapSplitsToTablets(tableName, tableId, tabLocator, splitsTodo);
 
-        List<Future<List<Text>>> splitTasks = new ArrayList<>();
+      ArrayList<Pair<TFateId,List<Text>>> opids = new ArrayList<>(tabletSplits.size());
 
-        for (Entry<KeyExtent,List<Text>> splitsForTablet : tabletSplits.entrySet()) {
-          Callable<List<Text>> splitTask = createSplitTask(tableName, splitsForTablet);
-          splitTasks.add(executor.submit(splitTask));
+      final ByteBuffer EMPTY = ByteBuffer.allocate(0);
+
+      // begin the fate operation for each tablet without waiting for the operation to complete
+      for (Entry<KeyExtent,List<Text>> splitsForTablet : tabletSplits.entrySet()) {
```

Comment:
Based on the top level comment I made, if we really wanted to speed this up then the RPCs being executed in this loop could be done in a thread pool.  Does not have to be done in this PR, these changes already speed things up.  Running RPCs in a thread pool would speed things up even more.